src/topology/order/lower_topology.lean

```diff
@@ -0,0 +1,406 @@
+      id.def], exact rfl, },
+  { rw [equiv.coe_trans, function.comp_app, equiv.prod_congr_apply, prod_map, equiv.coe_refl,
+      id.def], exact rfl, }
+end
+
+-- Why doesn't equiv.refl_symm work?
+@[simp] lemma refl_symm : (equiv.refl (α × β)).symm = @equiv.refl (α × β) := rfl
+
+section partial_order
+
+variables [partial_order α] [partial_order β] [topological_space α] [s : lower_topology α]
+  [topological_space β] [t : lower_topology β]
+
+include s t
+
+lemma prod_basis_is_topological_basis :
+  is_topological_basis
+  (image2 prod (lower_topology.lower_basis α) (lower_topology.lower_basis β)) :=
+is_topological_basis.prod lower_topology.is_topological_basis lower_topology.is_topological_basis
+
+instance [order_bot α] [order_bot β] : lower_topology (α × β) :=
+{ topology_eq_lower_topology :=
+  begin
+    rw le_antisymm_iff,
+    split,
+    { apply le_generate_from,
+    intros,
+    simp only [prod.exists, mem_set_of_eq] at H,
+    cases H with a,
+    cases H_h with b,
+    rw [← H_h_h, ← upper_set.coe_Ici, is_open_compl_iff, upper_set.prod_Ici],
+    apply is_closed.inter,
+    { apply is_closed.prod is_closed_univ,
+      rw upper_set.coe_Ici,
+      apply lower_topology.is_closed_Ici, },
+    { apply is_closed.prod, rw upper_set.coe_Ici,
+      apply lower_topology.is_closed_Ici,
+      apply is_closed_univ, } },
+    { rw (prod_basis_is_topological_basis α β).eq_generate_from,
+      apply le_generate_from,
+      intros U hU,
+      simp only [mem_image2, exists_and_distrib_left] at hU,
+      cases hU with V,
+      cases hU_h with hV,
+      cases hU_h_right with W,
+      cases hU_h_right_h with hW hUVW,
+      rw ← hUVW,
+      rw [lower_topology.lower_basis, mem_set_of_eq] at hV,
+      cases hV with F₁,
+      rw [lower_topology.lower_basis, mem_set_of_eq] at hW,
+      cases hW with F₂,
+      rw [← hV_h.2, ← hW_h.2, ← lower_set.coe_prod, upper_closure_compl_prod_upper_closure_compl,
+        lower_set.coe_inf],
+      apply is_open.inter,
+      { rw [upper_set.coe_compl, is_open_compl_iff, upper_closure_prod, upper_closure_univ,
+          ← upper_set.Ici_bot, ← upper_closure_singleton, ← upper_closure_prod],
+        convert lower_topology.is_closed_upper_closure _ _,
+        sorry,
+        apply set.finite.prod (finite_singleton ⊥) hW_h.1, },
+      { rw  [upper_set.coe_compl, is_open_compl_iff, upper_closure_prod, upper_closure_univ,
+          ← upper_set.Ici_bot, ← upper_closure_singleton, ← upper_closure_prod],
+        convert lower_topology.is_closed_upper_closure _ _,
+        sorry,
+       apply set.finite.prod hV_h.1 (finite_singleton ⊥), }, },
+  end }
+
+/--
+The lower topology of the partially ordered space α × β is homeomorphic to the product topology of
+the lower topology of the partially ordered space α with the lower topology of the partially ordered
+space β
+-/
+def lower_topology_prod_hom [order_bot α] [order_bot β] :
+  with_lower_topology (α × β) ≃ₜ (α × β) :=
```

Comment:
```suggestion
def with_lower_topology_iso [order_bot α] [order_bot β] :
  with_lower_topology α ≃ₜ α :=
```

Reply:
Why `iso`? Isn't `≃ₜ` a homeomorphism?

Reply:
My result is the statement that the product topology and the lower topology on a product space are homeomorphic, whereas I think your statement is that the lower topology from `with_lower_topology` is homeomorphic to the lower topology from `lower_topology`?

Reply:
@YaelDillies I've added the result that the mixin and the type synonym are homeomorphic, if that's what you wanted?

Reply:
That's better. Now you can just get rid of `lower_topology_prod_hom`

Reply:
Thanks. This was done here: https://github.com/leanprover-community/mathlib/pull/17426/commits/4521e64860ec6a6b14a37814132d3f640548c755